// @inworld/nodejs-sdk/build/src/entities/character.entity.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Character = void 0;
const util = require("node:util");
class Character {
    constructor(props) {
        this.id = props.id;
        this.resourceName = props.resourceName;
        this.displayName = props.displayName;
        this.assets = props.assets;
    }
    getId() {
        return this.id;
    }
    getResourceName() {
        return this.resourceName;
    }
    getDisplayName() {
        return this.displayName;
    }
    getAssets() {
        return this.assets;
    }
}
exports.Character = Character;
Character.prototype.getId = util.deprecate(Character.prototype.getId, 'getId() is deprecated. Use `id` property instead.');
Character.prototype.getDisplayName = util.deprecate(Character.prototype.getDisplayName, 'getDisplayName() is deprecated. Use `displayName` property instead.');
Character.prototype.getResourceName = util.deprecate(Character.prototype.getResourceName, 'getResourceName() is deprecated. Use `resourceName` property instead.');
Character.prototype.getAssets = util.deprecate(Character.prototype.getAssets, 'getAssets() is deprecated. Use `assets` property instead.');
