// @inworld/nodejs-sdk/build/src/entities/emotion-strength.entity.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EmotionStrength = exports.EmotionStrengthCode = void 0;
const util = require("node:util");
const packets_pb_1 = require("../../../proto/packets_pb");
var EmotionStrengthCode;
(function (EmotionStrengthCode) {
    EmotionStrengthCode["UNSPECIFIED"] = "UNSPECIFIED";
    EmotionStrengthCode["WEAK"] = "WEAK";
    EmotionStrengthCode["STRONG"] = "STRONG";
    EmotionStrengthCode["NORMAL"] = "NORMAL";
})(EmotionStrengthCode = exports.EmotionStrengthCode || (exports.EmotionStrengthCode = {}));
class EmotionStrength {
    constructor(strength) {
        this.code = strength;
    }
    isWeak() {
        return this.code === EmotionStrengthCode.WEAK;
    }
    isStrong() {
        return this.code === EmotionStrengthCode.STRONG;
    }
    static fromProto(code) {
        switch (code) {
            case packets_pb_1.EmotionEvent.Strength.UNSPECIFIED:
                return EmotionStrengthCode.UNSPECIFIED;
            case packets_pb_1.EmotionEvent.Strength.WEAK:
                return EmotionStrengthCode.WEAK;
            case packets_pb_1.EmotionEvent.Strength.STRONG:
                return EmotionStrengthCode.STRONG;
            case packets_pb_1.EmotionEvent.Strength.NORMAL:
                return EmotionStrengthCode.NORMAL;
        }
    }
}
exports.EmotionStrength = EmotionStrength;
EmotionStrength.prototype.isWeak = util.deprecate(EmotionStrength.prototype.isWeak, 'isWeak() is deprecated. Use code property instead.');
EmotionStrength.prototype.isStrong = util.deprecate(EmotionStrength.prototype.isStrong, 'isStrong() is deprecated. Use code property instead.');
