// @inworld/nodejs-sdk/build/src/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.status = exports.SessionToken = exports.Session = exports.Scene = exports.InworldPacketType = exports.InworldPacket = exports.InworldConnectionService = exports.InworldClient = exports.InworlControlType = exports.getStudioHost = exports.getEngineHost = exports.EmotionStrengthCode = exports.EmotionStrength = exports.EmotionBehaviorCode = exports.EmotionBehavior = exports.Character = void 0;
/**
 * Copyright 2022 Theai, Inc. (DBA Inworld)
 *
 * Use of this source code is governed by the Inworld.ai Software Development Kit License Agreement
 * that can be found in the LICENSE.md file or at https://www.inworld.ai/sdk-license
 */
const moduleAlias = require('module-alias');
moduleAlias.addAliases({
    '@proto': `${__dirname}/../proto`,
    '@root': `${__dirname}../../../`,
});
const grpc_js_1 = require("@grpc/grpc-js");
Object.defineProperty(exports, "status", { enumerable: true, get: function () { return grpc_js_1.status; } });
const inworld_client_1 = require("./clients/inworld.client");
Object.defineProperty(exports, "InworldClient", { enumerable: true, get: function () { return inworld_client_1.InworldClient; } });
const config_1 = require("./common/config");
const character_entity_1 = require("./entities/character.entity");
Object.defineProperty(exports, "Character", { enumerable: true, get: function () { return character_entity_1.Character; } });
const emotion_behavior_entity_1 = require("./entities/emotion-behavior.entity");
Object.defineProperty(exports, "EmotionBehavior", { enumerable: true, get: function () { return emotion_behavior_entity_1.EmotionBehavior; } });
Object.defineProperty(exports, "EmotionBehaviorCode", { enumerable: true, get: function () { return emotion_behavior_entity_1.EmotionBehaviorCode; } });
const emotion_strength_entity_1 = require("./entities/emotion-strength.entity");
Object.defineProperty(exports, "EmotionStrength", { enumerable: true, get: function () { return emotion_strength_entity_1.EmotionStrength; } });
Object.defineProperty(exports, "EmotionStrengthCode", { enumerable: true, get: function () { return emotion_strength_entity_1.EmotionStrengthCode; } });
const inworld_packet_entity_1 = require("./entities/inworld_packet.entity");
Object.defineProperty(exports, "InworlControlType", { enumerable: true, get: function () { return inworld_packet_entity_1.InworlControlType; } });
Object.defineProperty(exports, "InworldPacket", { enumerable: true, get: function () { return inworld_packet_entity_1.InworldPacket; } });
Object.defineProperty(exports, "InworldPacketType", { enumerable: true, get: function () { return inworld_packet_entity_1.InworldPacketType; } });
const scene_entity_1 = require("./entities/scene.entity");
Object.defineProperty(exports, "Scene", { enumerable: true, get: function () { return scene_entity_1.Scene; } });
const session_entity_1 = require("./entities/session.entity");
Object.defineProperty(exports, "Session", { enumerable: true, get: function () { return session_entity_1.Session; } });
const session_token_entity_1 = require("./entities/session_token.entity");
Object.defineProperty(exports, "SessionToken", { enumerable: true, get: function () { return session_token_entity_1.SessionToken; } });
const inworld_connection_service_1 = require("./services/inworld_connection.service");
Object.defineProperty(exports, "InworldConnectionService", { enumerable: true, get: function () { return inworld_connection_service_1.InworldConnectionService; } });
const config = config_1.Config.getInstance();
const getStudioHost = () => config.getStudioHost();
exports.getStudioHost = getStudioHost;
const getEngineHost = () => config.getEngineHost();
exports.getEngineHost = getEngineHost;
